src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.Targets.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Symbols
+{
+    public sealed partial class ModuleInitializersTests
+    {
+        [Fact]
+        public void IgnoredOnLocalFunction()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+class C
+{
+    internal static void M()
+    {
+        LocalFunction();
+        [ModuleInitializer]
+        static void LocalFunction() { }
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics(
+                // (9,10): error CS8793: A module initializer must be an ordinary method
```

Comment:
consider instead enumerating kinds of methods that are "non-ordinary". "A module initializer cannot be a local function, constructor, destructor, operator, or accessor".

Reply:
Cool. Should there be test of some sort to prevent us from forgetting to grow this list when AttributeTargets.Method begins to apply to new things, e.g. lambdas? It wasn't long ago that local functions wouldn't have been in this list.

Reply:
@AlekseyTs suggests the terminology 'an ordinary member method' rather than enumerating kinds, due to the length of the list.